test/test_nn.py

```diff
@@ -7223,6 +7223,30 @@ def test_cosine_embedding_loss_margin_no_reduce(self):
                          loss_reference_fns['CosineEmbeddingLoss'](input1, input2, target,
                                                                    margin=0.5, reduction='none'))
 
+    def test_cosine_embedding_loss_with_diff_type(self):
+        for device in device_():
+            for dt1 in torch.testing.get_all_math_dtypes(device):
+                for dt2 in torch.testing.get_all_math_dtypes(device):
+                    for dt3 in torch.testing.get_all_math_dtypes(device):
+                        if dt3 == torch.uint8:
+                            continue
+                        input1 = torch.tensor([[2, 3, 4], [6, 2, 4]], dtype=dt1, device=device)
+                        input2 = torch.tensor([[2, 3, 5], [3, 2, 1]], dtype=dt2, device=device)
+                        target = torch.tensor([1, -1], dtype=dt3, device=device)
+                        result = torch.nn.functional.cosine_embedding_loss(input1, input2, target)
+                        self.assertEqual(result.item(), 0.4672, 0.001)
```

Comment:
instead of a magic number, why don't you compare with the case where everything is a double.

Reply:
Changed